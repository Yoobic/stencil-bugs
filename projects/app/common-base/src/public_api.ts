export { AppCommonBaseModule } from './lib/app-common-base.module';

export * from './lib/bases/app-chat-page/app-chat-page.component';
export * from './lib/bases/app-entity-action-page/app-entity-action-page.component';
export * from './lib/bases/app-form-dynamic-page/app-form-dynamic-page.component';
export * from './lib/bases/app-grid-dialog-page/app-grid-dialog-page.component';
export * from './lib/bases/app-menu-page/app-menu-page.component';
export * from './lib/bases/app-root/app-root.component';
export * from './lib/bases/app-search-page/app-search-page.component';
export * from './lib/bases/app-walkthrough-page/app-walkthrough-page.component';
export * from './lib/bases/app-whatsnew-page/app-whatsnew-page.component';
export * from './lib/bases/base-page/base-page.component';

export * from './lib/components/app-profile/app-profile.component';
export * from './lib/components/app-login/app-login.component';
export * from './lib/components/app-grid/app-grid.component';
export * from './lib/components/app-form-dynamic/app-form-dynamic.component';

export * from './lib/services/dialog/dialog.service';
export * from './lib/services/utils/utils.service';

export * from './lib/animations/animations';
export * from './lib/animations/transitions';

export * from './lib/modals/app-result-dialog-page/app-result-dialog-page.component';