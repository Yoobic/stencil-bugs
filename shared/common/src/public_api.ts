export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';