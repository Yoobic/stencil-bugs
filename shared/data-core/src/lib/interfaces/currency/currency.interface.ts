import { Model } from '../../decorators/model/model.decorator';
import { Searchable } from '../../decorators/searchable/searchable.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { FormFieldType, ICurrency } from '@shared/interfaces';
import { Translate } from '@shared/translate';

@Model({
    className: 'Currency',
    collectionName: 'currencies',
    fields: ['*'],
    include: []
})

export class Currency extends ICurrency {

    @Editable('Currency', { required: true, type: FormFieldType.autocomplete, title: 'LANGUAGE', translate: true, values: Translate.availablesLanguage, clearable: false })
    @Searchable('Currency') currency: string;

    @Editable('Currency', { required: true, type: FormFieldType.text })
    @Searchable('Currency') symbol: string;

    @Editable('Currency', { required: true, type: FormFieldType.number, title: 'RATE1' })
    @Searchable('Currency') rate: number;

}
