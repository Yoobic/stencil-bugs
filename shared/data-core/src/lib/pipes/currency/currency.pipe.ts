import { Pipe, PipeTransform } from '@angular/core';
import { DecimalPipe } from '@angular/common';
import { Session } from '../../services/session/session.service';
import { Translate } from '@shared/translate';
import { filter } from 'lodash-es';

@Pipe({ name: 'currency' })
export class CurrencyPipe implements PipeTransform {

    private translate: Translate;
    private decimalPipe: DecimalPipe;

    constructor(translate: Translate, private session: Session) {
        this.translate = translate;
        this.decimalPipe = new DecimalPipe(this.translate.getCurrentAngularLocale());
    }

    transform(value: number, ...args: string[]): any {
        let precision = args && args.length > 0 ? args[0] : 2;

        if (!value && value !== 0) {
            return '';
        }
        if (Math.abs(value) >= 1 || value === 0) {
            precision = precision || 0;
        } else {
            precision = precision || 1;
        }
        let symbol = this.translate.get('CURRENCY');
        let rates = filter(this.session.currencies, c => c.symbol === symbol);

        let rate = 1;
        if (rates.length > 0) {
            rate = rates[0].rate;
        }

        let displayValue = this.decimalPipe.transform(value * rate, '1.0-' + precision.toString());

        if (symbol === 'Â£') {
            return symbol + ' ' + displayValue;
        } else {
            return displayValue + ' ' + symbol;
        }
    }
}
