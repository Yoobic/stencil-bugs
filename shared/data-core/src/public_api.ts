export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
