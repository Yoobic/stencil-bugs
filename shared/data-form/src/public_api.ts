
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';