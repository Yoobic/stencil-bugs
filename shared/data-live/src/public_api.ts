
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';