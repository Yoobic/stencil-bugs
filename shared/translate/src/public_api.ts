export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';