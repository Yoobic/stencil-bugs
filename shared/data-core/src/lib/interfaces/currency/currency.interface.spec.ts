import { Currency } from './currency.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: Currency', () => {
    it('should create an entity', () => {
        let entity = new Currency();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('Currency');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});