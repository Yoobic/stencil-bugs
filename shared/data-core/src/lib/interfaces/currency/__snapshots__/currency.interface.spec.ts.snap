// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: Currency should create an entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "clearable": false,
      "filterName": "currency",
      "name": "currency",
      "required": true,
      "title": "LANGUAGE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "en",
        "us",
        "fr",
        "de",
        "es",
        "nl",
        "pl",
        "it",
        "ru",
        "zhs",
        "zht",
        "pt",
        "kr",
        "ja",
        "ua",
        "he",
        "ar",
        "cz",
        "th",
        "tr",
        "bg",
        "el",
        "sl",
        "sk",
      ],
    },
    Object {
      "filterName": "symbol",
      "name": "symbol",
      "required": true,
      "title": "SYMBOL",
      "type": "text",
    },
    Object {
      "filterName": "rate",
      "name": "rate",
      "required": true,
      "title": "RATE1",
      "type": "number",
    },
  ],
  "appareance": Map {},
  "className": "Currency",
  "collectionName": "currencies",
  "feathersService": undefined,
  "fields": Array [
    "*",
  ],
  "include": Array [],
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [
    "currency",
    "symbol",
    "rate",
  ],
  "type": [Function],
}
`;
